### PR TITLE
BPR-113 Handle nested dependencies

### PR DESCRIPTION
**Pull request by RealFoXsTeRDOTA**
Had some issues related to self-dependencies and usings in components that had no dependencies so attempted to fix those too

image
image

Green to indicate 'Open'
Red to indicate 'Closed'
... yes, we should change it to make the arrow dotted instead eventually

### CODE REVIEW

File: BPR.Tests/AnalysisServiceTests.cs [hunk 1]
```diff
@@ -36,123 +39,197 @@ public class AnalysisServiceTests {
...
-        var result = AnalysisService.GetDependencyAnalysisOnComponent(usingList, component);
-        foreach (var dependency in component.Dependencies) {
-            result.AddRange(AnalysisService.GetDependencyAnalysisOnComponent(usingListDependency,dependency));
-        }
-        //Assert
-        Assert.That(result, !Is.Empty);
-        Assert.That(result[0].Severity,Is.EqualTo(ViolationSeverity.Major));
-        Assert.That(result[0].Type,Is.EqualTo(ViolationType.ForbiddenDependency));
+    private static AnalysisArchitecturalComponent GenerateDummyComponentWithNoDependencies()
+    {
```
Comment (stefaniatomuta) [inline BPR.Tests/AnalysisServiceTests.cs:219]: would it make sense to have this separated from the test in another class?

**Comment by stefaniatomuta** [comment]
Not related to this PR, but why do we have Global usings in the test library?

**Comment by RealFoXsTeRDOTA** [comment]
> Not related to this PR, but why do we have Global usings in the test library?

Could argue that if there ever was a reason to actually use global usings, a test project is definitely the place as every file (except files that hold test data or so) uses the test and mocking framework. I don't see why we would remove it tbh.

File: BPR.Tests/AnalysisServiceTests.cs (hunk 1 above)
Comment (RealFoXsTeRDOTA) [reply BPR.Tests/AnalysisServiceTests.cs:219]: Yeah, will do so.

**Review by stefaniatomuta — APPROVED** [review]

**Comment by RealFoXsTeRDOTA** [comment]
@vladdirlea2002 hi 😃

File: BPR.Tests/TestData.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                {
+                    Name = "BPRBlazor"
+                }
+            },
+            Dependencies = new()
+            {
+            }
+        };
+    }
+}
```
Comment (stefaniatomuta) [inline BPR.Tests/TestData.cs:104]: 👍

**Review by stefaniatomuta — APPROVED** [review]

**Pull request merged**